sdk/database/helper/connutil/sql.go

```diff
@@ -12,15 +12,33 @@ import (
 	"sync"
 	"time"
 
+	"github.com/hashicorp/go-uuid"
+
 	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/go-secure-stdlib/parseutil"
+	"github.com/mitchellh/mapstructure"
+
 	"github.com/hashicorp/vault/sdk/database/dbplugin"
 	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
-	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	authTypeIAM      = "gcp_iam"
+	dbTypePostgres   = "pgx"
+	dbTypeMSSQL      = "mssql"
+	cloudSQLPostgres = "cloudsql-postgres"
+	cloudSQLMSSQL    = "cloudsql-sqlserver"
 )
 
 var _ ConnectionProducer = &SQLConnectionProducer{}
 
+var (
+	driversMu sync.RWMutex
+	drivers   = make(map[string]cloudSQLCleanup)
```

Comment:
What's the reason for these being at the package level? Would it be possible to move them to `SQLConnectionProducer`?

Reply:
Drivers in go are registered globally and registering a driver with a colliding name is a panic, so anything that thinks about registering a driver needs to either put it somewhere where it only happens once (the usual way is in `init()` of the library implementing the driver) or a singleton way to check if a driver might have already been registered.

It turns out we need to multi-registered essentially the same driver if we're using the GCP library (the driver is attached to the IAM identity, more or less), so we have an alternate method using GUIDs that obsoletes this approach, but that's why the map was at the package level, rather than instanced.

Reply:
It seems odd that `drivers` is never used, so I'm not sure I understand how this can guard against a colliding name/panic. `SQLConnectionProducer.Init` has a boolean named `Initialized`, so can that be used as the singleton check to ensure that Register is called once, i.e. `if !c.Initialized { // register the driver }`?

Reply:
Ah okay, thanks for the explanation @kpcraig . Since we're using uuids for the driver name, are we safe to remove this code block?